### PR TITLE
Understanding Support Vector Machine(SVM) algorithm

### LINKED ISSUE #4055: [Machine Learning] Understanding Support Vector Machine(SVM) algorithm
#### Proposed title of article
Understanding Support Vector Machine(SVM) algorithm 

#### Proposed article introduction
This supervised machine learning approach, known as the Support Vector Machine (SVM), can be used to tackle problems in classification and regression. Classification problems are the most common application for this technique. N is the number of features you have, and the value of each component is the value of a specific coordinate. 

The SVM algorithm plots each data point as a point in n-dimensional space. This tutorial will talk about Support Vector Machine(SVM) algorithm in detail.

#### Key takeaways 
By the end of this tutorial the reader will have understood the following;

- What is Support Vector Machine?
- How does it work?
- How to implement SVM in Python and R?
- How to tune Parameters of SVM?
- Pros and Cons associated with SVM

#### Article quality
My article will be approachable for readers, and my code examples will be clear and concise so that they can be understood by a wide range of people.
The Support Vector Machine(SVM) algorithm will be demonstrated in my article by creating an actual file, which will make the tutorial more practical and more straightforward to grasp.

#### References
N/A

### PR DESCRIPTION
**Pull request by jmuchina**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This fixes #4055

### CODE REVIEW

**Comment by srishilesh** [comment]
@jmuchina 

Welcome to EngEd program!

Congratulations on your first article.

Before we begin with the review, could you please add the PR description back? And, please ensure if you followed all the checks mentioned there. (If so, please check those boxes).

Let's wait for the CMs to get this PR, plagiarism checked, before we can start with the actual review.

Thanks.

**Comment by ahmadmardeni1** [comment]
Please always use the template provided by us.
I edited your initial comment and added the template. Check what matches your article.

**Comment by ahmadmardeni1** [comment]
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.
[Understanding Support Vector Machine(SVM) algorithm#4908](https://github.com/section-engineering-education/engineering-education/files/7533417/New.Scan.8_25.AM.Copyleaks.Report.pdf)

**Comment by jmuchina** [comment]
Hello, @ahmadmardeni1 
I have made the required changes
Thank You.

**Comment by srishilesh** [comment]
@jmuchina 

Great.

Let's get started with the review.

Meanwhile, please add a GitHub repo link containing the full code, after the conclusion.

Also, please have a look at the guidelines for authors information. There was an update to it, 2 days ago.

Ping me when these are done. I'll get started with the preliminary review.

Thanks.

**Comment by srishilesh** [comment]
@jmuchina 

The formatting looks good.

Just make sure to perform a Grammarly check, if you haven't.

Also, rename the directory and image paths for `content/articles/understanding-support-vector-machine-smv-algorithm/index.md` (smv -> svm)

And, don't forget the GitHub URL containing full code.

Ping me when the changes are done.

**Comment by srishilesh** [comment]
Just an update.

I'll complete the next round of review by the end of the day.

**Comment by srishilesh** [comment]
@jmuchina 

Can you ping me (@srishilesh ) via Slack?

File: content/articles/understanding-support-vector-machine-algorithm/index.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+To do this, we first find the hyperplane that adequately separates the two categories of information as shown in the image below:
+
+![The diagram shows an SVM example](/engineering-education/understanding-support-vector-machine-algorithm/vector.jpg)
+
+Each observation's coordinates are used to construct support vectors for the model. As a consequence, the classification algorithm serves as a boundary between the two classes.
+
+### How does Support Vector Machine work?
+#### Determining hyperplane
+Three hyper-planes may be seen in this picture (A, B, and C).
```
Comment (srishilesh) [inline content/articles/understanding-support-vector-machine-algorithm/index.md:41]: Firstly, what's a hyperplane?

File: content/articles/understanding-support-vector-machine-algorithm/index.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+That's why it goes through a long list of transformations before we decide how to segment the data using our labels or outputs.
+
+![Original hyper-plane input](/engineering-education/understanding-support-vector-machine-algorithm/caseeight.jpg)
+
+### Implementing SVM in Python
+Scikit-learn, a Python package for machine learning algorithms, is widely used.
+
+Using SVM is as simple as importing the library, creating an object, fitting a model, and making a prediction.
+
+Using a scenario and dataset to better understand how SVM operates for classification will be the next step.
```
Comment (srishilesh) [inline content/articles/understanding-support-vector-machine-algorithm/index.md:111]: Where's the dataset?

File: content/articles/understanding-support-vector-machine-algorithm/index.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+clf = SVC(kernel='a linear')
+clf.fit(q_train,w_train)
+y_pred = clf.pred(q_trial)
+print(accurate_marks(w_test,w_predict))
+```
+
+**Explanation** We used `kernel=a linear` in the previous code to generate an SVM for variables that can be segregated linearly. On the other hand, nonlinear data might want a different strategy. As a result, we used the classifier to identify patterns in the original training data before moving on.
+
+### Tuneing Parameters in SVM
```
Comment (srishilesh) [inline content/articles/understanding-support-vector-machine-algorithm/index.md:125]: Typo

File: content/articles/understanding-support-vector-machine-algorithm/index.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+When using machine learning methods, model accuracy may be improved by modifying parameter values. Look at the list of SVM parameters.
+```
+sklearn.svm.SVC(C=1.0, kernel='rbf', degree=2, gamma=0.0, coef0=0.0, shrinking=True, probability=False,tol=0.01, cache_size=150, class_quantity=None, verbose=False, maximum_iters=-1, random_nature=None)
+```
+
+We'll discuss `kernel,` `gamma,` and `C,` three crucial model performance parameters.
+
+ A variety of options are available from the kernel in this situation, including `linear,` `rbf,` and `poly` (the default being rbf ). It is preferable to employ the notions of `rbf` and `poly` for nonlinear hyperplane issues. Tuneing of parameters is used in a linear kernel to classify two iris data set characteristics based on their kind.
```
Comment (srishilesh) [inline content/articles/understanding-support-vector-machine-algorithm/index.md:133]: Where are we discussing these?

File: content/articles/understanding-support-vector-machine-algorithm/index.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+y_pred = clf.pred(q_trial)
+print(accurate_marks(w_test,w_predict))
+```
+
+**Explanation** We used `kernel=a linear` in the previous code to generate an SVM for variables that can be segregated linearly. On the other hand, nonlinear data might want a different strategy. As a result, we used the classifier to identify patterns in the original training data before moving on.
+
+### Tuneing Parameters in SVM
+When using machine learning methods, model accuracy may be improved by modifying parameter values. Look at the list of SVM parameters.
+```
+sklearn.svm.SVC(C=1.0, kernel='rbf', degree=2, gamma=0.0, coef0=0.0, shrinking=True, probability=False,tol=0.01, cache_size=150, class_quantity=None, verbose=False, maximum_iters=-1, random_nature=None)
```
Comment (srishilesh) [inline content/articles/understanding-support-vector-machine-algorithm/index.md:128]: Explain

File: content/articles/understanding-support-vector-machine-algorithm/index.md [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+Anyway, we should not be too concerned about the present polynomial kernel. Due to efficiency difficulties, this isn't as handy as you may imagine for (both computational and forecasting). Instead of using the Radial Basis Function kernel, use a linear SVM (or logistic regression) for basic problems. This is the general principle to keep in mind.
+
+The RBF kernel SVM's decision area is also linear. As a consequence, the RBF core SVM elevates the samples into a high-dimensional feature set with linear decision boundaries that may be used to differentiate classes.
+
+### Conclution
+ As evidenced in this tutorial, we have looked more into the Support Vector Machine machine learning algorithm. We discussed the model's working concept, how to construct it in Python, how to optimize its parameters, and its benefits and disadvantages.
+ 
+Here is the full code [repo link](https://gist.github.com/mblondel/586753/f740949d0336484567dd422fe53445ac8821f5b2) of Support Vector Machines.
```
Comment (srishilesh) [inline content/articles/understanding-support-vector-machine-algorithm/index.md:163]: Was this the code that you used in the tutorial?

It's a totally different code

File: content/articles/understanding-support-vector-machine-algorithm/index.md [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+
+Anyway, we should not be too concerned about the present polynomial kernel. Due to efficiency difficulties, this isn't as handy as you may imagine for (both computational and forecasting). Instead of using the Radial Basis Function kernel, use a linear SVM (or logistic regression) for basic problems. This is the general principle to keep in mind.
+
+The RBF kernel SVM's decision area is also linear. As a consequence, the RBF core SVM elevates the samples into a high-dimensional feature set with linear decision boundaries that may be used to differentiate classes.
+
+### Conclution
```
Comment (srishilesh) [inline content/articles/understanding-support-vector-machine-algorithm/index.md:160]: Typo

**Review by srishilesh — CHANGES_REQUESTED** [review]
@jmuchina 

On performing a tech review, your PR lacks quality and clarity in explaining concepts.

Moreover, the code snippets are not explained properly. You have just stated the facts, and not explained "why" or "how"

The GitHub URL is totally a different code.

We will be closing this PR to allow you to make necessary changes. Ping me in the new PR, when it's ready.

Make sure that your new PR must not require any additional revisions, and must be ready to publish article.

Thanks.

**Comment by srishilesh** [comment]
@jmuchina 

Do we have any updates on this?

Let me know if you need help in creating a new PR.

Thanks.
